libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/MainMenu.tsx

```diff
@@ -183,23 +187,46 @@ export class MainMenu extends React.PureComponent<
     );
   };
 
-  private toggleCohortSettingsPanel = (): void =>
+  private toggleCohortSettingsPanel = (): void => {
+    if (!this.state.cohortSettingsPanelVisible) {
+      this.onButtonClick(TelemetryEventName.MainMenuCohortSettingsClick);
+    }
     this.setState((prev) => ({
       cohortSettingsPanelVisible: !prev.cohortSettingsPanelVisible
     }));
+  };
 
-  private toggleDashboardSettings = (): void =>
+  private toggleDashboardSettings = (): void => {
+    if (!this.state.dashboardSettingsVisible) {
+      this.onButtonClick(
+        TelemetryEventName.MainMenuDashboardConfigurationClick
+      );
+    }
     this.setState((prev) => ({
       dashboardSettingsVisible: !prev.dashboardSettingsVisible
     }));
+  };
+
   private toggleChangeCohortVisibility = () => {
+    if (!this.state.changeCohortVisible) {
+      this.onButtonClick(TelemetryEventName.MainMenuSwitchCohortClick);
+    }
     this.setState((prev) => ({
       changeCohortVisible: !prev.changeCohortVisible
     }));
   };
   private toggleCreateCohortVisibility = () => {
+    if (!this.state.createCohortVisible) {
+      this.onButtonClick(TelemetryEventName.MainMenuNewCohortClick);
+    }
     this.setState((prev) => ({
       createCohortVisible: !prev.createCohortVisible
     }));
   };
+  private onButtonClick = (eventName: TelemetryEventName) => {
```

Comment:
Since this is just logging, should we rename this is method? onButtonClick might seem like we do some action here.

Reply:
Great point!. Updated